### PR TITLE
Convert IOErrors into keycert errors. Raise ssh errors from KeyCertErrors

### PR DESCRIPTION
**Pull request by adiroiban**
Problem
=======

Ran into the following exception when issuing a server command from an unprivileged account:

```
  File "C:\Program Files (x86)\Server\lib\lib\site-packages\chevah\serv
er\commons\commands.py", line 608, in subcommand_generate_ssh_key
    exit_code, message, key = crypto_generate_ssh_key(options)
  File "C:\Program Files (x86)\Server\lib\lib\site-packages\chevah\keyc
ert\ssh.py", line 135, in generate_ssh_key
    with open_method(_path(private_file), 'wb') as file_handler:
IOError: [Errno 13] Permission denied: u'id_rsa'
```

Changes
=======

Catch all exceptions and turn them into own execeptions

As a drive by I have reorganized ssh exceptions to raise exceptions inherited from a single exception type.

How to test
=========

reviewers: @alibotean 

check that changes make sense

Test pass on Travis-ci ... but I think that they fail on windows... I have manually sent the changes to buildbot

Thanks!

### CODE REVIEW

**Comment by adiroiban** [comment]
needs-review

File: chevah/keycert/exceptions.py [hunk 1]
```diff
@@ -14,3 +14,18 @@ def __init__(self, message):
 
     def __str__(self):
         return self.message.encode('utf-8')
+
+
+class BadKeyError(KeyCertException):
+    """
+    Raised when a key isn't what we expected from it.
+
+    XXX: we really need to check for bad keys
```
Comment (alibotean) [inline chevah/keycert/exceptions.py]: Missing ticket id.

**Comment by adiroiban** [comment]
buildbot is broken and I will fix it later (tm) :(

**Comment by alibotean** [comment]
changes-approved

**Pull request merged**